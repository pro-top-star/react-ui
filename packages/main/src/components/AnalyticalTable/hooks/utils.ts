export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
