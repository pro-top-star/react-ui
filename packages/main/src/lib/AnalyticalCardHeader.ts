import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
