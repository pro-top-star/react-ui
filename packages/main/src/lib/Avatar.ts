import { Avatar } from '../webComponents/Avatar';

export { Avatar };
