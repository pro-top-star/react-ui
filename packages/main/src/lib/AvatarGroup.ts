import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
