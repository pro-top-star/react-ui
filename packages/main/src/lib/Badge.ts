import { Badge } from '../webComponents/Badge';
import type { BadgePropTypes } from '../webComponents/Badge';

export { Badge };
export type { BadgePropTypes };
