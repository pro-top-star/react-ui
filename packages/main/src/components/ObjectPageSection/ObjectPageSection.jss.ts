import { ThemingParameters } from '@ui5/webcomponents-react-base/lib/ThemingParameters';

const styles = {
  header: {
    padding: '0 3rem 0 2.5rem',
    borderBottom: `1px solid ${ThemingParameters.sapGroup_TitleBorderColor}`,
    boxSizing: 'border-box',
    height: '2.75rem'
  },
  title: {
    height: '2.75rem',
    color: ThemingParameters.sapTextColor,
    fontSize: ThemingParameters.sapFontHeader4Size,
    borderBottom: `1px solid ${ThemingParameters.sapActiveColor}`,
    display: 'inline-block',
    maxWidth: '100%',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
    fontFamily: ThemingParameters.sapFontFamily
  },
  uppercase: {
    textTransform: 'uppercase'
  },
  sectionContent: {
    whiteSpace: 'normal'
  },
  sectionContentInner: {
    padding: '1rem 2rem 2rem 2.5rem',
    fontFamily: ThemingParameters.sapFontFamily
  }
};

export default styles;
