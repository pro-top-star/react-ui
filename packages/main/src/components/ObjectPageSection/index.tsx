import { StyleClassHelper } from '@ui5/webcomponents-react-base/lib/StyleClassHelper';
import { useConsolidatedRef } from '@ui5/webcomponents-react-base/lib/useConsolidatedRef';
import { usePassThroughHtmlProps } from '@ui5/webcomponents-react-base/lib/usePassThroughHtmlProps';
import { useScrollElement } from '@ui5/webcomponents-react-base/lib/useScrollElement';
import React, { FC, forwardRef, ReactNode, ReactNodeArray, RefObject } from 'react';
import { createUseStyles } from 'react-jss';
import { CommonProps } from '../../interfaces/CommonProps';
import { EmptyIdPropException } from '../ObjectPage/EmptyIdPropException';
import styles from './ObjectPageSection.jss';

export interface ObjectPageSectionPropTypes extends CommonProps {
  title?: string;
  id: string;
  titleUppercase?: boolean;
  children: ReactNode | ReactNodeArray;
}

const useStyles = createUseStyles<keyof ReturnType<typeof styles>>(styles, { name: 'ObjectPageSection' });

const ObjectPageSection: FC<ObjectPageSectionPropTypes> = forwardRef(
  (props: ObjectPageSectionPropTypes, ref: RefObject<any>) => {
    const { title, id, children, titleUppercase, className, style, tooltip } = props;
    const classes = useStyles();

    if (!id) {
      throw new EmptyIdPropException('ObjectPageSection requires a unique ID property!');
    }

    const sectionRef: RefObject<HTMLElement> = useConsolidatedRef(ref);
    const htmlId = `ObjectPageSection-${id}`;

    useScrollElement(htmlId, sectionRef, {
      spy: true
    });

    const titleClasses = StyleClassHelper.of(classes.title);
    if (titleUppercase) {
      titleClasses.put(classes.uppercase);
    }

    const passThroughProps = usePassThroughHtmlProps(props);

    return (
      <section
        ref={sectionRef}
        role="region"
        className={className}
        style={style}
        title={tooltip}
        {...passThroughProps}
        id={htmlId}
      >
        <div role="heading" className={classes.header}>
          <div className={titleClasses.valueOf()}>{title}</div>
        </div>
        {/* TODO Check for subsections as they should win over the children */}
        <div className={classes.sectionContent}>
          <div className={classes.sectionContentInner}>{children}</div>
        </div>
      </section>
    );
  }
);

ObjectPageSection.defaultProps = {
  title: '',
  // @ts-ignore
  isSection: true,
  titleUppercase: true
};

ObjectPageSection.displayName = 'ObjectPageSection';

export { ObjectPageSection };
