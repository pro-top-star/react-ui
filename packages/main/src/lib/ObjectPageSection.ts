import { ObjectPageSection } from '../components/ObjectPageSection';

export { ObjectPageSection };
