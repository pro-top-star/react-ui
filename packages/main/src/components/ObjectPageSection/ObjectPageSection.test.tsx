import { createPassThroughPropsTest, renderThemedComponent } from '@shared/tests/utils';
import { ObjectPageSection } from '@ui5/webcomponents-react/lib/ObjectPageSection';
import { shallow } from 'enzyme';
import React from 'react';

describe('ObjectPageSection', () => {
  test('Renders with children', () => {
    const wrapper = renderThemedComponent(
      <ObjectPageSection id={'1'} title="Test" titleUppercase>
        This is my Text
      </ObjectPageSection>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('ObjectPage w/ lowercase title', () => {
    const wrapper = renderThemedComponent(
      <ObjectPageSection id={'1'} title="Test" titleUppercase={false}>
        This is my Text
      </ObjectPageSection>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Empty Id Prop', () => {
    // @ts-ignore
    const renderer = () => shallow(<ObjectPageSection.InnerComponent />);
    expect(renderer).toThrow();
  });

  createPassThroughPropsTest(ObjectPageSection);
});
