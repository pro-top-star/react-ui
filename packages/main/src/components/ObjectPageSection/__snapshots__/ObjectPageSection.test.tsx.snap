// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectPageSection ObjectPage w/ lowercase title 1`] = `
<section
  data-component-name="ObjectPageSection"
  id="ObjectPageSection-1"
  role="region"
>
  <div
    class="ObjectPageSection-header-0"
    role="heading"
  >
    <div
      class="ObjectPageSection-title-0"
    >
      Test
    </div>
  </div>
  <div
    class="ObjectPageSection-sectionContent-0"
  >
    <div
      class="ObjectPageSection-sectionContentInner-0"
    >
      This is my Text
    </div>
  </div>
</section>
`;

exports[`ObjectPageSection Renders with children 1`] = `
<section
  data-component-name="ObjectPageSection"
  id="ObjectPageSection-1"
  role="region"
>
  <div
    class="ObjectPageSection-header-0"
    role="heading"
  >
    <div
      class="ObjectPageSection-title-0 ObjectPageSection-uppercase-0"
    >
      Test
    </div>
  </div>
  <div
    class="ObjectPageSection-sectionContent-0"
  >
    <div
      class="ObjectPageSection-sectionContentInner-0"
    >
      This is my Text
    </div>
  </div>
</section>
`;
